src/api.ts

```diff
@@ -36,19 +37,28 @@ export async function synchronize(
 ): Promise<OperationResult[]> {
   log('Synchronize');
 
-  const params: SynchronizeParams = {
+  let params: SynchronizeParams = {
     force: false,
     'skip-migration': false,
     ...options
   };
   const { onStarted: _, ...invokeParams } = params;
 
+  const migrationPath = getMigrationPath(config);
+  const dirExist = await existDirectory(migrationPath);
   // TODO: Need to preload the SQL source code under this step.
   const { knexMigrationConfig } = await init.prepare(config, {
     loadSqlSources: true,
     loadMigrations: !params['skip-migration']
   });
```

Comment:
@Soneshaps Add a `dirExists` flag here as well and return null from `resolveMigrationContext` incaase of truth.